src/devtools.ts

```diff
@@ -25,18 +29,56 @@ chrome.devtools.inspectedWindow.eval(
   function(isShopifyStore: boolean) {
     if (isShopifyStore) {
       chrome.devtools.panels.create('Shopify', '', './devtools.html');
-      const toolbar = new Toolbar();
 
       if (getBrowserTheme() === 'dark') {
         document.documentElement.setAttribute('data-theme', 'dark');
       }
 
-      toolbar.refreshButton.addEventListener('click', refreshPanel);
-      toolbar.zoomOutButton.addEventListener('click', zoomOutFlamegraph);
+      addEventListenerToButtons();
     }
   },
 );
 
+function addEventListenerToButtons() {
+  const toolbar = new Toolbar();
+
+  toolbar.refreshButton.addEventListener('click', refreshPanel);
+  toolbar.zoomOutButton.addEventListener('click', zoomOutFlamegraph);
+
+  document
+    .querySelector(selectors.searchButton)!
+    .addEventListener('click', function(event) {
+      event.preventDefault();
+      search();
+    });
+
+  document
+    .querySelector(selectors.clearButton)!
+    .addEventListener('click', function(event) {
+      event.preventDefault();
+      clear();
+    });
+}
+
+function search() {
+  const searchParam = (document.querySelector(
+    selectors.searchParam,
+  ) as HTMLInputElement).value;
+
+  if (typeof liquidFlamegraph.flamegraph !== 'undefined') {
+    liquidFlamegraph.flamegraph.search(escape(searchParam));
```

Comment:
Do I need to escape since we don't show this anywhere?

Reply:
I would escape it since we are searching through the flamegraph nodes

Reply:
Where is `liquidFlamegraph.flamegraph.search` defined? is this a undocumented functionality from the Flame graph lib?

Reply:
I also did not find it in their doc. I was able to find here through their [example](http://martinspier.io/d3-flame-graph/).

The source code for this is [here](https://github.com/spiermar/d3-flame-graph/blob/gh-pages/index.html).

Reply:
lol okay